File: qqrobot-spring-boot-starter/src/main/java/de/honoka/qqrobot/starter/config/AdminWebConfig.java
package de.honoka.qqrobot.starter.config;

import de.honoka.qqrobot.starter.component.admin.AdminLoginInterceptor;
import de.honoka.qqrobot.starter.config.property.AdminProperties;
import jakarta.annotation.Resource;
import org.jetbrains.annotations.NotNull;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.Arrays;

@Configuration
public class AdminWebConfig implements WebMvcConfigurer {

    @Resource
    private AdminLoginInterceptor adminLoginInterceptor;

    @Override
    public void addInterceptors(@NotNull InterceptorRegistry registry) {
        String prefix = AdminProperties.WEB_PREFIX;
        registry.addInterceptor(adminLoginInterceptor)
                .addPathPatterns(prefix + "/**")
                .excludePathPatterns(Arrays.asList(
                        prefix + "/static/**",
                        prefix + "/api/login",
                        prefix + "/favicon.ico",
                        prefix + "/index.html",
                        prefix + "/"
                ));
    }

    @Override
    public void addResourceHandlers(@NotNull ResourceHandlerRegistry registry) {
        registry.addResourceHandler(AdminProperties.WEB_PREFIX + "/**")
                .addResourceLocations("classpath:/web/admin/");
    }
}

File: qqrobot-spring-boot-starter/src/main/java/de/honoka/qqrobot/starter/config/AdminWebConfig.kt
package de.honoka.qqrobot.starter.config

import de.honoka.qqrobot.starter.component.admin.AdminLoginInterceptor
import de.honoka.sdk.spring.starter.core.context.springBean
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@EnableConfigurationProperties(AdminProperties::class)
@Configuration
class AdminWebConfig : WebMvcConfigurer {
    
    private val excludePathPatterns = run {
        val prefix = AdminProperties.WEB_PREFIX
        listOf(
            "$prefix/static/**",
            "$prefix/api/login",
            "$prefix/favicon.ico",
            "$prefix/index.html",
            "$prefix/"
        )
    }
    
    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.run {
            addInterceptor(AdminLoginInterceptor::class.springBean).run {
                addPathPatterns("${AdminProperties.WEB_PREFIX}/**")
                excludePathPatterns(excludePathPatterns)
            }
        }
    }
    
    override fun addResourceHandlers(registry: ResourceHandlerRegistry) {
        registry.run {
            addResourceHandler("${AdminProperties.WEB_PREFIX}/**").run {
                addResourceLocations("classpath:/web/admin/")
            }
        }
    }
}

@ConfigurationProperties("honoka.qqrobot.admin")
data class AdminProperties(
    
    /**
     * 后台管理界面的登录密码
     */
    var password: String = "123456"
) {
    
    companion object {
        
        const val WEB_PREFIX = "/admin"
    }
}
